pkg/gds/certman/certs.go

```diff
@@ -675,3 +680,147 @@ func (c *CertificateManager) getCertStorage() (path string, err error) {
 	log.Warn().Str("certs", path).Msg("using a temporary directory for cert downloads")
 	return path, err
 }
+
+// HandleCertificateReissuance iterates through each VASP in the database and checks
+// if their identity certificate will be expiring soon, sending reminder email within
+// the 30 and 7 day checkpoints if so, and reissuing the identity certificate 10 days before
+// expiration.
+func (c *CertificateManager) HandleCertificateReissuance() {
+	var (
+		err            error
+		expirationDate time.Time
+	)
+
+	// Iterate through the VASPs in the database.
+	vasps := c.db.ListVASPs()
+	for vasps.Next() {
+		var vasp *pb.VASP
+		if vasp, err = vasps.VASP(); err != nil {
+			log.Error().Err(err).Msg("could not parse vasp request from database")
+			continue
+		}
+
+		// Calculate the number of days before the VASP's certificate expires.
+		notAfter := vasp.IdentityCertificate.NotAfter
+		if expirationDate, err = time.Parse(time.RFC3339, notAfter); err != nil {
+			log.Error().Err(err).Msg(fmt.Sprintf("could not parse %s's cert reissuance date", vasp.Id))
+		}
+		reissuanceDate := expirationDate.Add(-time.Hour * 240) // Calculate the reissuance date, 10 days before expiration
+		timeBeforeExpiration := time.Until(expirationDate)
+
+		switch daysBeforeReissuance := timeBeforeExpiration.Hours() / 24; {
+		// Seven days before expiration, send a cert reissuance reminder to VASP.
+		case daysBeforeReissuance <= 7:
+			c.email.SendContactReissuanceReminder(vasp, 7, reissuanceDate)
+
+		// Ten days before expiration, reissue the VASP's identity certificate, send the email with the created pkcs12
+		// password and send the whisper link, as well as notifying the TRISA admin that reissuance has started.
+		case daysBeforeReissuance <= 10:
+			c.reissueIdentityCertificates(vasp)
```

Comment:
As the PR currently stands, I don't think anything is preventing the reissuance process from triggering multiple times before the other certman loop handles it. I think we need to check if there's an existing certificate request in `READY_TO_SUBMIT` on the VASP to avoid the double reissue. If we need to tackle this in a separate story that's fine, since it probably requires writing a helper method to do the check.

Reply:
I created a story for this in the automated reissuance epic.